source/developers/how-tos/get-ready-for-python-dev.rst

```diff
@@ -0,0 +1,82 @@
+
+The most basic way of creating a virtualenv is to use the builtin ``venv`` module in Python::
+
+  python3.8 -m venv .venv-3.8
+
+This example creates a virtualenv at ``./.venv-3.8/``. Each time you will be working with Python in the repo, you'll need to activate the virtualenv for that terminal session::
+
+  source .venv-3.8/bin/activate
+
+Once you have done this, any calls to ``python``, ``pip``, or other Python executables will refer to the ones managed by the virtualenv.
+
+.. note::
+
+   If you create virtualenvs inside repos, you'll need to tell git to ignore them. The easiest way to do this is to create a file called ``.gitignore-global`` in your home directory and add the line ``.venv-3.*/``. Alternatively, you can create the virtualenvs elsewhere in your filesystem.
+
+Many developers use wrapper scripts (or write their own using shell aliases). One commonly used tool is ``virtualenvwrapper``, which manages the virtualenv directories outside of repositories; this avoids several issues with git and other tools being able to see the virtualenv, but will require explicitly naming each virtualenv.
+
+Working on a Repo
+=================
+
+#. Make sure you are on the default branch and have the latest version of the code::
+
+     git checkout main
+     git pull
+
+   Some repos instead have ``master`` as their default branch.
+
+#. Activate your virtualenv, as described above.
+#. Install or update the Python requirements for the repo into your virtualenv::
+
+     make requirements
+
+   In a few repos, this might also install NodeJS requirements, or the Makefile target might have a different name.
+
+#. Optionally, ensure you can run tests::
+
+     make test
+
+   Depending on the repository this might run not just unit tests, but also linters and other quality checks. The Makefile may also define ``make quality`` for code quality checks, or even a comprehensive ``test-all`` target.
+
+#. Make a branch for your changes::
+
+     git checkout -b <your_github_username>/<short_descriptive_label>
+
+#. As you change code and add tests, you can use ``make test`` to check if tests are still passing.
+#. Run ``make test`` one more time and commit your changes with ``git commit``. Follow the `conventional commits`_ documentation. Make sure your commit message is informative and describes why the change is being made. While the first line of the message should be terse, the body of the message has plenty of room for details.
+#. Push your changes to GitHub. If you have write access to the repository, you can run ``git push``. If not, you'll need to fork the repository in GitHub. After forking, you can use ``git remote add fork <fork_url>`` to tell git about the fork, and then ``git push fork`` to push your branch to your fork.
+#. In GitHub, open a pull request for your changes and ask for a review.
```

Comment:
```suggestion
#. In GitHub, open a pull request (PR). In the PR description, include anything that could help reviewers understand and test your change.
#. If you have a repository maintainer in mind, you can reach out to them and ask for review. Otherwise, an Open edX project manager will triage your PR and find someone to review it. This process can take some time; we appreciate your patience!
```

"Ask for review" only works for folks who know maintainers/CCs. Without getting too deep into "How to make a PR", here are a couple items that could clue them into the next steps.